### ISSUE TITLE
My two cents on the discussion + the problem with reshape

### ISSUE THREAD

**Issue by smartinsightsfromdata** [matloff/TidyverseSkeptic#18]
I enjoyed Prof. Matloff post, as usual fair and thought provoking.
(apologies if I take a bit of time to get to the point)

Disclaimer: I am as much as a tidyverse skeptic as prof Matloff (possibly a tidyverse basher!).

**A personal journey**
I learned R in 2013, when (thanks to the gods) the tidyverse had not emerged.
I found R a fun programming language, full of nice little time-saving idiosyncrasies (the apply family, do.call function etc.).
Rapidly the need to speed let me discover and learn (a bit painfully, I confess) data.tables.
But once the efforts paid off, like for anything you conquer with your own efforts, I felt a great sense of satisfaction.
I started to work progressively in python (now almost exclusively).  In parallel this all-devouring monster of the tidyverse started to emerge and now risks to take over this much belowed language.

**The nature of the tidyverse**
Programming languages have always oscillated between two extremes:

- something very verbose, easy for newbies and not concise (think Cobol)
- something very concise, hard to learn but so expressive (think APL, not surprisingly considered an inspiration for R)

I think is personal choice what to prefer. I do prefer programming in APL-like languages and R.

The tidyverse is *perversly* making R looking more like Cobol (perverse as in changing the nature of).

This would not be so bad if the tidyverse would keep itself separate. I found this not been the case and certainly one of the crunching points is `reshape`.

As we all know, reshape (or is subsequent re-incarnation reshape2) helps in changing a table in wide to long format and viceversa.  Doing so it computes also clever aggregations (sum of course plus others).  

Two functions in one statement? Heresy against the fundamental principles of the tidyverse!

Bottom line: reshape2 has not been updated since 2017 and is described as "archived" (certianly not very good for any corporate programming policies).

It is no surprise that the data.table team has just come up with dedicated functions for reshaping long and wide (they existed before as extension of reshape, but certainly they could see thet they were operating on shaky ground).

I mentioned above as I'm using python more and more.  Python is less *fun* than R, certainly.
But in my personal opinion is better than a tidyverse, Cobol-like no-fun world.

It is just sad, really.

**Comment by drag05** [comment]
@smartinsightsfromdata : The lack of updates of  `reshape2` hasn't *forced* me to choose `tidyverse`: I became an exclusive `data.table` user dropping all things "tidy".  I think `data.table` solves many `tidy` messes in one big swoop.

**Comment by matloff** [comment]
:-)
